core/thread/inc/ROOT/TThreadedObject.hxx

```diff
@@ -38,22 +40,15 @@ namespace ROOT {
     * \brief Defines the number of threads in some of ROOT's interfaces.
     */
    struct TNumSlots {
-      int fVal; // number of slots
+      unsigned int fVal; // number of slots
```

Comment:
`unsigned` arithmatics (`++`, `<=`) are a nightmare. Do we really need this?

Reply:
> Do we really need this?

As much as you ever really need unsigned integers...it's nice that `TNumSlots{-1}` simply does not compile and I don't have to add a ctor that checks for non-negativity. Plus the number of slots must be non-negative, always grows, and should stay quite far from UINT_MAX. What kind of problems do you expect?

(just let me know if you want an `int`)

Reply:
OK. But comparison etc is evil - the general (committee) recommendation is "don't use unsigned unless you do bit manipulations". https://wesmckinney.com/blog/avoid-unsigned-integers/ https://www.boost.org/doc/libs/develop/libs/safe_numerics/doc/html/tutorial/3.html https://github.com/ericniebler/stl2/issues/182 :-) Your call, I get your argument.